### PR TITLE
refactor: esm build

### PR DESCRIPTION
**Pull request by pimlie**

### CODE REVIEW

File: scripts/rollup.config.js [hunk 1]
```diff
@@ -84,5 +87,35 @@ export default [
...
+    external: Object.keys(pkg.dependencies)
+  }),
+  // browser esm build
+  rollupConfig({
+    input: 'src/browser.js',
+    output: {
+      file: pkg.web.replace('.js', '.esm.browser.js'),
+      format: 'es'
+    },
+    external: Object.keys(pkg.dependencies)
```
Comment (pimlie) [inline scripts/rollup.config.js]: Not sure if this makes sense, but the deepmerge dependency doesnt export a esm build so rollup cant include it (due to this webpack bug: https://github.com/webpack/webpack/issues/6584):
```
[!] Error: 'default' is not exported by node_modules/deepmerge/dist/umd.js
https://rollupjs.org/guide/en#error-name-is-not-exported-by-module-
src/shared/merge.js (1:7)
1: import deepmerge from 'deepmerge'
          ^
2: import { findIndex } from '../utils/array'
3: import { applyTemplate } from './template'
Error: 'default' is not exported by node_modules/deepmerge/dist/umd.js
```

Comment (galvez) [reply scripts/rollup.config.js]: @pimlie you might need https://github.com/nuxt/fabula/blob/master/build/builtins.js

Comment (galvez) [reply scripts/rollup.config.js]: Also see `rollup.config.js` there.

Comment (pimlie) [reply scripts/rollup.config.js]: @galvez Thx, but deepmerge is an external package not a builtin one. If you copy this config (which I stole mostly from pi0) you might want to consider removing the external stuff for the esm browser builds so all dependencies are inlined. Although to be honest I am not sure what exactly the conventions are for esm browser builds

**Pull request merged**